### PR TITLE
:zap: Batch plot_2d primitives via LineCollection

### LINKED ISSUE #44: Refactor plot routine to draw Voronoi edges and cell boundaries simultaneously using batch plotting
Currently, the `_plot_routine` method in `finite_voronoi.py` first loops through and draws all Voronoi ridge segments (background edges) and then iterates over each cell to draw its boundary one-by-one. This results in two separate passes for plotting and individual calls per cell.

Matplotlib provides tools (such as LineCollection, PolyCollection, etc.) that allow batch plotting of multiple lines or polygons in a single, more efficient call. By restructuring the code, it should be possible to prepare all coordinates for both the background Voronoi edges and the cell boundaries in advance and then plot them together (simultaneously), potentially improving efficiency and making the code cleaner.

**Proposal:**
- Refactor the plotting logic in `_plot_routine` so that all background Voronoi ridge segments and cell boundaries are drawn in a single batch operation, making use of suitable Matplotlib collections or array-based plotting.
- This may require constructing lists or arrays of all line segments and polygons prior to the actual plotting calls.
- Evaluate potential performance improvements and ensure plot appearance does not regress.

**Benefits:**
- Potential speed-up for large systems with many cells.
- Cleaner and more maintainable plotting code.
- More consistent visual style, as all primitives can be drawn in unified style calls.

**Acceptance Criteria:**
- All Voronoi edges and all cell boundaries are drawn more efficiently, using batch drawing routines.
- Plot appearance remains equivalent to the current implementation (no significant regression).
- Code is easier to read and maintain.

### PR DESCRIPTION
**Pull request by wwang721**
## Summary

- Replace per-ridge and per-edge `ax.plot` loops in `_plot_routine` with batched `LineCollection` calls.
- Background Voronoi ridges: split into finite (solid) and formerly-infinite (dashed) collections.
- Cell boundaries: flatten ragged per-cell lists once, build a cell-wrapping next-vertex index, then emit one collection for straight edges (zorder=1) and one for arcs + full-circle degenerate cells (zorder=2).
- All kwargs (colors, widths, alpha, explicit zorder=1/2) preserved exactly; no new zorder added where main left it unset.

Closes #44.

## Test plan

- [x] Visual diff vs. main via `plt.savefig` — identical PNG output.
- [x] `pytest tests/test_geom.py tests/test_calibrate.py`
- [x] Check rendering for edge cases: `N=1`, `N=2` (both separated and overlapping), `N>=3` with extension vertices, cells with no straight edges (all-arc), cells with no arcs (all-straight).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pyafv/finite_voronoi.py [hunk 1]
```diff
@@ -881,15 +881,20 @@ def _plot_routine(self, ax: matplotlib.axes.Axes, vor: Voronoi, vertices_all: np
...
+            rv_orig = np.asarray(vor.ridge_vertices)
+            if rv_orig.size > 0:
+                segs = vertices_all[np.asarray(ridge_vertices_all, dtype=int)]  # (R,2,2)
+                inf_mask = (rv_orig == -1).any(axis=1)
+                finite_segs = segs[~inf_mask]
+                dashed_segs = segs[inf_mask]
+                if len(finite_segs):
+                    ax.add_collection(LineCollection(finite_segs, colors='k', linestyles='-', linewidths=0.5, alpha=line_alpha))
+                if len(dashed_segs):
+                    ax.add_collection(LineCollection(dashed_segs, colors='k', linestyles='--', linewidths=0.5, alpha=line_alpha))
```
[the remark this answers is not in the document]
Comment (wwang721) [reply pyafv/finite_voronoi.py:897]: These are the background Voronoi edges, which should be underneath.
